aten/src/ATen/core/TensorImpl.h

```diff
@@ -9,24 +9,115 @@
 #include "ATen/core/TensorTypeIdRegistration.h"
 #include "ATen/core/LegacyTypeDispatch.h"
 #include "ATen/core/Backend.h"
+#include "ATen/core/context_base.h"
 
+#include "caffe2/core/allocator.h"
+#include "caffe2/core/common.h"
+#include "caffe2/core/flags.h"
 #include "caffe2/core/logging.h"
 
-struct THTensor;
+// A global boolean variable to control whether we free memory when a Tensor
+// is shrinked to a smaller size. As a result, a Tensor is always going to
+// keep the memory allocated for its maximum capacity reshaped to so far.
+CAFFE2_DECLARE_bool(caffe2_keep_on_shrink);
```

Comment:
No, and IMO it would violate our API contract to do so. So it's only respected by the Caffe2 resizing functions. Working out the relationship between the PyTorch resize and the Caffe2 resize is some detailed work that we have to do.

I suppose it's possible to refactor this so that the gflags are in Tensor, and then rewrite the internal implementations in TensorImpl to take them in as booleans, thereby ejecting the gflags dependency from TensorImpl. (Let me know if you want me to do this.) But long term, if we really want to axe these we'll have to do some sort of ablation study where we kill the flag and see if anything in prod breaks.

Reply:
Similarly, Caffe2 has another method ShrinkTo which also doesn't respect this flag.